language/move-lang/src/expansion/ast.rs

```diff
@@ -46,12 +46,24 @@ pub struct Script {
 pub struct ModuleDefinition {
     pub loc: Loc,
     pub is_source_module: bool,
+    pub friends: UniqueMap<ModuleIdent, Friend>,
```

Comment:
Let me break out a quick PR for UniqueSet. I've hit it a bit with abilities and this is another candidate for having it

Reply:
Actually, I think you want UniqueMap<ModuleIdent, Loc> as the loc wrapping the module ident might not be the loc where the friend was declared

Reply:
Right, I think that was also the reason why I have this weird
```
#[derive(PartialEq, Clone, Debug)]
pub enum Friend_ {
    Ok,
    UnresolvedError,
}
pub type Friend = Spanned<Friend_>;
```

To keep the separate `Loc` (hidden behind `Spanned` though...). But I'll have `Friend_` removed.